inabox/tests/setup_disperser_harness.go

```diff
@@ -942,3 +969,218 @@ func startController(
+	apiServerLogger.Info("Creating meterer")
+
+	mtConfig := meterer.Config{
+		ChainReadTimeout: 5 * time.Second,
+		UpdateInterval:   1 * time.Second, // Match deploy config for tests
+	}
+
+	paymentChainState, err := meterer.NewOnchainPaymentState(ctx, chainReader, apiServerLogger)
+	if err != nil {
+		_ = logFile.Close()
+		return "", fmt.Errorf("failed to create onchain payment state: %w", err)
+	}
+	if err := paymentChainState.RefreshOnchainPaymentState(ctx); err != nil {
+		_ = logFile.Close()
+		return "", fmt.Errorf("failed to make initial query to the on-chain state: %w", err)
+	}
+
+	// Use the standard v2 payment table prefix
+	const v2PaymentPrefix = "e2e_v2_"
+	meteringStore, err := meterer.NewDynamoDBMeteringStore(
+		awsConfig,
+		v2PaymentPrefix+"reservation",        // ReservationsTableName
+		v2PaymentPrefix+"ondemand",           // OnDemandTableName
+		v2PaymentPrefix+"global_reservation", // GlobalRateTableName
+		apiServerLogger,
+	)
+	if err != nil {
+		_ = logFile.Close()
+		return "", fmt.Errorf("failed to create offchain store: %w", err)
+	}
+
+	mt := meterer.NewMeterer(
+		mtConfig,
+		paymentChainState,
+		meteringStore,
+		apiServerLogger,
+	)
+	mt.Start(ctx)
+
+	// Create server config
+	serverConfig := disperser.ServerConfig{
+		GrpcPort:              "32005", // Use a specific port for API server
+		GrpcTimeout:           10 * time.Second,
+		MaxConnectionAge:      5 * time.Minute,
+		MaxConnectionAgeGrace: 30 * time.Second,
+		MaxIdleConnectionAge:  1 * time.Minute,
+	}
+
+	metricsConfig := disperser.MetricsConfig{
+		HTTPPort:      "9100",
+		EnableMetrics: true,
+	}
+
+	// Max number of symbols per blob (based on typical config)
+	const maxNumSymbolsPerBlob = 16 * 1024 * 1024
+
+	// Onchain state refresh interval
+	onchainStateRefreshInterval := 1 * time.Second
+
+	// Create API server v2
+	// Note: meterer is nil when using controller-mediated payments, otherwise it's the legacy meterer
+	apiServerV2, err := apiserver.NewDispersalServerV2(
+		serverConfig,
+		blobStore,
+		metadataStore,
+		chainReader,
+		mt, // meterer (nil when using controller-mediated payments, otherwise legacy meterer)
+		authenticator,
+		kzgCommitter,
+		maxNumSymbolsPerBlob,
+		onchainStateRefreshInterval,
+		apiServerLogger,
+		metricsRegistry,
+		metricsConfig,
+		false, // ReservedOnly
+		config.TestConfig.UseControllerMediatedPayments,
+		controllerAddress,
+	)
+	if err != nil {
+		_ = logFile.Close()
+		return "", fmt.Errorf("failed to create API server v2: %w", err)
+	}
+
+	// Start API server in background
+	go func() {
+		apiServerLogger.Info("Starting API server v2", "port", serverConfig.GrpcPort, "logFile", logFilePath)
+		if err := apiServerV2.Start(ctx); err != nil {
+			apiServerLogger.Error("API server v2 failed", "error", err)
+		}
+	}()
+
+	// Store server in harness
+	harness.APIServerV2 = apiServerV2
+
+	actualAddress := fmt.Sprintf("localhost:%s", serverConfig.GrpcPort)
```

Comment:
actualAddress as opposed to what? Why not just call it `listeningAddress` or `grpcAddress` or something?